### PR TITLE
feat(nodejs): introduce safer timestamp API

### PR DESCRIPTION
**Pull request by puzpuzpuz**
**Note**. This is a breaking change, so the patch bumps the version to 2.0.0.

Adds time unit as an optional parameter to timestamp-accepting methods:
```js
  const bday = Date.parse('1856-07-10');
  sender
    .table('inventors')
    .symbol('born', 'Austrian Empire')
    .timestampColumn('birthday', bday, 'ms') // notice 'ms' here
    .intColumn('id', 0)
    .stringColumn('name', 'Nicola Tesla')
    .at(Date.now(), 'ms'); // notice 'ms' here
```

The unit defaults to 'us' (native QuestDB resolution) in both `timestampColumn()` and `at()`. Supported values are:
* 'ns' - nanoseconds
* 'us' - microseconds
* 'ms' - milliseconds

Also, includes some code style changes (e.g. consistent `'` instead of mixed `'` and `"`) and a number of improvements, including example unification with the Java client.

### Considered alternatives

[Temporal](https://tc39.es/proposal-temporal/docs/) types could be used instead of the optional parameter, but the thing is that the spec is still experimental, so it's too early to rely on it.

### CODE REVIEW

File: test/sender.test.js [hunk 1]
```diff
@@ -402,117 +536,108 @@ describe('Sender message builder test suite (anything not covered in client inte
...
         expect(
-            () => sender.table("tableName")
-                .timestampColumn("intField", 123.222)
-        ).toThrow("Value must be an integer or BigInt, received 123.222");
+            () => sender.table('tableName')
+                .timestampColumn('intField', 123.222)
+        ).toThrow('Value must be an integer or BigInt, received 123.222');
     });
 
     it('throws exception if designated timestamp is not a string or bigint', function () {
```
Comment (glasstiger) [inline test/sender.test.js:565]: nit: name of the test should change too. string -> integer

File: test/sender.test.js [hunk 2]
```diff
@@ -402,117 +536,108 @@ describe('Sender message builder test suite (anything not covered in client inte
...
-                .symbol("name", "value")
-                .at("")
-        ).toThrow("Empty string is not allowed as designated timestamp");
+            () => sender.table('tableName')
+                .symbol('name', 'value')
+                .at(23232322323.05)
+        ).toThrow('Designated timestamp must be an integer or BigInt, received 23232322323.05');
     });
 
     it('throws exception if designated timestamp is invalid', function () {
```
Comment (glasstiger) [inline test/sender.test.js:574]: this test now looks the same as the previous one.
maybe we could change this to pass a string value.
```
.at('invalid_dts')
```

File: test/sender.test.js (hunk 1 above)
Comment (puzpuzpuz) [reply test/sender.test.js:565]: Fixed in 0a4380e11d58161e405e6d14fb50b32d4f75c1ff

File: test/sender.test.js (hunk 2 above)
Comment (puzpuzpuz) [reply test/sender.test.js:574]: Good catch. Fixed in 0a4380e11d58161e405e6d14fb50b32d4f75c1ff

**Review by glasstiger — APPROVED** [review]

**Pull request merged**